### DISCUSSION TITLE
Does dropping a row from the `objects` table in Postgres delete the object?

### DISCUSSION THREAD

**Discussion by typedrat** [supabase/supabase#13131] [Questions]
I'm curious just how deep the integration goes between the database and the storage API. If I reference a storage object from a row in another table through a foreign key and use a trigger on that table to `DELETE` objects that are orphaned, will that also get rid of the underlying file in the bucket?

**Comment by GaryAustin1** [comment]
NO.  You have to make a storage-api call to delete the actual file data from s3 storage.

The guides in docs all now show an example of doing this with the http extension... https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs#storage-management

You are also probably better off storing the bucket (if not fixed) and pathname to the file versus the id in a foreign key.
